core/src/main/java/org/apache/accumulo/fate/AdminUtil.java

```diff
@@ -390,17 +388,23 @@ private FateStatus getTransactionStatus(ReadOnlyTStore<T> zs, Set<Long> filterTx
 
       zs.unreserve(tid, 0);
 
-      if ((filterTxid != null && !filterTxid.contains(tid))
-          || (filterStatus != null && !filterStatus.contains(status)))
-        continue;
-
-      statuses.add(new TransactionStatus(tid, status, debug, hlocks, wlocks, top, timeCreated));
+      if (includeByStatus(status, filterStatus) && includeByTxid(tid, filterTxid)) {
```

Comment:
This line in the added test expects that no arguments prints all
```
ts.exec("fate -print", true, "txid:", true);
```
This line in the added test expects that no txid but a status return results (on match):
```
ts.exec("fate -print -t IN_PROGRESS", true, "txid:", true);
```
The case of just a txid (success) is untested, but verified manually because getting the txid was not straight forward.

I thought the discussion centered around the `includeByTxid && includeByStatus` and if that should be an or.  The current behavior in this code is that if you specify txid and a status. both must match. (and both take a list)

Reply:
> I thought the discussion centered around the includeByTxid && includeByStatus and if that should be an or. The current behavior in this code is that if you specify txid and a status. both must match. (and both take a list)

Yes.

I am going to take a look at the new Test today and see if I can come up with the untested case as well.